applications/xcm-tools.md

```diff
@@ -0,0 +1,162 @@
+|     6. | Pull request     | Create pull request merge into [substrate-api-rpc](https://github.com/itering/substrate-api-rpc) and [scale.go](https://github.com/itering/scale.go) |  
+
+### Milestone 2
+
+* **Estimated duration:** 1.5 month
+* **FTE:**  1
+* **Costs:** 10000 USDC
+
+| Number | Deliverable       | Specification                                                                                         |
+|-------:|-------------------|-------------------------------------------------------------------------------------------------------|
+|    0a. | License           | MIT                                                                                                   |
+|    0b. | Documentation     | Simple documentation on how to use and how to test                                                    |
+|     1. | Send ump message  | support [UMP](https://wiki.polkadot.network/docs/learn-xcm#vmp-vertical-message-passing) message send |  
+|     2. | Send dmp message  | support [DMP](https://wiki.polkadot.network/docs/learn-xcm#vmp-vertical-message-passing) message send |  
+|     3. | Send hrmp message | support [HRMP](https://wiki.polkadot.network/docs/learn-xcm#hrmp-xcmp-lite) message send              |  
+|     4. | Github action     | Auto test when commit code                                                                            |  
+|     5. | Testing           | Including all the unit tests mentioned above                                                          |
+
+### Milestone 3
+
+* **Estimated duration:** 4 week
+* **FTE:**  1
+* **Costs:** 8000 USDC
+
+| Number | Deliverable            | Specification                                                                          |
+|-------:|------------------------|----------------------------------------------------------------------------------------|
+|    0a. | License                | MIT                                                                                    |
+|    0b. | Documentation          | Simple documentation on how to use and how to test                                     |
+|     1. | Parse xcm instructions | Parse instructions raw data, support xcm format v1 & v2                                |  
+|     2. | Tracing transaction    | Support xcm format v1 & v2                                                             |  
+|     3. | Cli support            | Add command line tool to send message & parse xcm instructions && tracking transaction |  
+|     4. | Github action          | Auto test when commit code                                                             |  
+|     5. | Testing                | Including all the unit tests mentioned above                                           |
+
+## Future Plans
+
+XCM v3 will be released this year, and I will support xcm v3 later.
```

Comment:
v3 is [already merged](https://github.com/paritytech/polkadot/pull/4097), could you integrate its support right away?

Reply:
OK, let me take a look. I don't have a deep understanding of xcm v3 yet, but I am willing to support xcm v3 together.